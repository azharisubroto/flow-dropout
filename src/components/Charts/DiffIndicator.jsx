import PropTypes from 'prop-types'
import IconButton from '@/components/IconButton'
import { IoArrowDownCircleOutline, IoArrowUpCircleOutline } from 'react-icons/io5'

/**
 * Diff Indicator
 * @param {Number} diffValue - Number of differential value
 */
const DiffIndicator = ({ diffValue }) => {
  return (
    <div className="diff-container">
      <div>
        <IconButton size={50}>
          {parseInt(diffValue) >= 0 ? <IoArrowUpCircleOutline size={25} /> : <IoArrowDownCircleOutline size={25} />}
        </IconButton>
      </div>

      <div className="break"></div>

      {/* Diff number */}
      <div className="diff">{diffValue ? diffValue : 0}%</div>

      <style jsx>{`
        .diff-container {
          display: flex;
          align-items: center;
          justify-content: center;
          max-width: 50px;
          flex-flow: row wrap;
        }

        .break {
          width: 100%;
        }

        .diff {
          color: #9b9b9b;
          text-align: center;
          font-size: 14px;
          position: relative;
          bottom: -10px;
        }
      `}</style>
    </div>
  )
}

DiffIndicator.propTypes = {
  /** - Value number of diff */
  diffValue: PropTypes.number
}

export default DiffIndicator
