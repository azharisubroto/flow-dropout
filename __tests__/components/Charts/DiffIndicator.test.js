import React from 'react'
import { shallow } from 'enzyme'
import DiffIndicator from '@/components/Charts/DiffIndicator'

describe('<DiffIndicator />', () => {
  let wrapper, DiffIndicatorComp
  const diff_val = 80

  beforeEach(() => {
    DiffIndicatorComp = <DiffIndicator diffValue={diff_val} />

    wrapper = shallow(DiffIndicatorComp)
  })

  it('should renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should contain correct classes', () => {
    expect(wrapper.find('.diff-container')).toHaveLength(1)
    expect(wrapper.find('.break')).toHaveLength(1)
    expect(wrapper.find('.diff')).toHaveLength(1)
  })

  it('should render correct diff value', () => {
    expect(wrapper.find('.diff').text()).toEqual('80%')
  })

  it('should render correct icon', () => {
    expect(wrapper.find('IoArrowUpCircleOutline')).toHaveLength(1)

    const IndicatorWrap = shallow(<DiffIndicator diffValue={-90} />)
    expect(IndicatorWrap.find('IoArrowDownCircleOutline')).toHaveLength(1)
  })
})
