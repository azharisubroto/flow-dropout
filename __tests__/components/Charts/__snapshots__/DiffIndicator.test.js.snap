// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiffIndicator /> should renders correctly 1`] = `ShallowWrapper {}`;
